### ISSUE TITLE
Laravel 5 routes don't resolve correctly

### ISSUE THREAD

**Issue by JonoB** [Codeception/Codeception#2000]
I've set up the Laravel 5 module, along with a few others. For some reason, the base url is not being constructed properly inside of the tests when using the `route()` function.

```yml
// api.suite.yml
class_name: ApiTester
modules:
    enabled: [Laravel5, REST, ApiHelper, Asserts]
    config:
        Laravel5:
            environment_file: .env.testing
```

```php
// .env.testing
APP_ENV=testing
APP_DEBUG=true
APP_KEY=whatever
APP_URL=http://fsadmin.dev

DB_DEFAULT=testing
```

```php
class AuthCest
{
    public function test_logout(ApiTester $i)
    {
        $i->assertEquals('testing', \App::environment()); // passes  
        $i->assertEquals('http://fsadmin.dev', \Config::get('app.url')); // passes

        $i->assertEquals('http://fsadmin.dev/api/auth/logout', route('api.auth.logout')); 

        // Failed asserting that two strings are equal.
        --- Expected
       +++ Actual
       -'http://fsadmin.dev/api/auth/logout'
       +'http://:/api/auth/logout#'
    }
}
```

Basically, all urls constructed from route are missing the base URL (fsadmin.dev), and I have no idea why.

Any help much appreciated.

**Comment by janhenkgerritsen** [comment]
The `app.php` config file contains the following comment for the `app.url` config variable:

```
    /*
	|--------------------------------------------------------------------------
	| Application URL
	|--------------------------------------------------------------------------
	|
	| This URL is used by the console to properly generate URLs when using
	| the Artisan command line tool. You should set this to the root of
	| your application so that it is used when running Artisan tasks.
	|
	*/
```

As you can see the `app.url` config variable is only used in the Artisan command line tool. The `route()` function retrieves the domain from the Request object. Since there is no request in your example the domain will be empty. But the results of your tests are as to be expected, since your assertions are wrong.

**Comment by JonoB** [comment]
That's a shame that named routes cannot be used in codeception in L5, because they used to work perfectly in L4 :(

**Comment by janhenkgerritsen** [comment]
Why can't you use the `amOnRoute` method of the Laravel 5 module?

Your example test is also not really useful, because you are only testing Laravel specific functionality, which you do not need to test, because you can assume it works as expected. Maybe you have a better example test case? For now I do not see a reason to "fix" this behavior, but if there is a good reason of course I can be convinced otherwise.

**Comment by CristianGiordano** [comment]
My pull request #2005 should resolve this,

**Comment by JonoB** [comment]
@janhenkgerritsen My assertions are not real tests - they are proof of concept to demonstrate my point.

The reason why resolving `route()` method correctly is shown in the following example:

```php
$url = route('api.auth.register');

$i->sendPOST($url, [
    'email' => $this->currentEmail,
    'password' => $this->userPassword,
]);
```

In this case, `$url` is not correct because the `route()` method cannot resolve correctly.

**Comment by janhenkgerritsen** [comment]
@JonoB You convinced me, I will look into this issue.

@CristianGiordano Your pull request only works if you call `route()` after a request has been made, for example after calls to `amOnPage()` or `amOnRoute()`. If you call `route()` before a request has been made you still get the existing behavior. 

I'll have to dig a bit deeper into the code to see what will be the best way to fix this. Will do that sometime this week.

**Comment by CristianGiordano** [comment]
Yes you do but surely people should use the API provided?

I will also take a look into the bootstrapping and see if I can resolve this. 

Thanks for your time :)

**Comment by JonoB** [comment]
Here's one way that we're playing with at the moment:

```php
    /**
     * Initialize the Laravel framework.
     *
     * @throws ModuleConfig
     */
    protected function initializeLaravel()
    {
        $this->app = $this->bootApplication();
        $baseUrl = env('APP_URL');
        $fakeRequest = Request::create($baseUrl);
        $this->app->instance('request', $fakeRequest);
        $this->client = new LaravelConnector($this->app);
        $this->client->followRedirects(true);
    }
```

**Comment by janhenkgerritsen** [comment]
I just pushed a commit that should fix this issue. This will be included in the release of Codeception 2.0.15.
